src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestExplainAction.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.resthandler
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.IndexStateManagementPlugin
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.elasticapi.getPolicyName
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.models.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.settings.ManagedIndexSettings
+import org.elasticsearch.action.admin.cluster.state.ClusterStateRequest
+import org.elasticsearch.action.admin.cluster.state.ClusterStateResponse
+import org.elasticsearch.action.support.IndicesOptions
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.rest.BaseRestHandler
+import org.elasticsearch.rest.BytesRestResponse
+import org.elasticsearch.rest.RestChannel
+import org.elasticsearch.rest.RestController
+import org.elasticsearch.rest.RestRequest
+import org.elasticsearch.rest.RestResponse
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.rest.action.RestBuilderListener
+
+class RestExplainAction(
+    settings: Settings,
+    controller: RestController
+) : BaseRestHandler(settings) {
+
+    companion object {
+        const val EXPLAIN_BASE_URI = "${IndexStateManagementPlugin.ISM_BASE_URI}/explain"
```

Comment:
Should this be ```_explain``` similar to Elasticsearch's API?

Reply:
Discussed this a bit with Drew. Idea is that since we already have namespace `_opendistro/_ism` to indicate that the API is under the ISM we don't really need more underscore in explain API.